usaspending_api/common/validator/award_filter.py

```diff
@@ -24,7 +24,22 @@
     {"name": "extent_competed_type_codes", "type": "array", "array_type": "text", "text_type": "search"},
     {"name": "keywords", "type": "array", "array_type": "text", "text_type": "search", "text_min": 3},
     {"name": "legal_entities", "type": "array", "array_type": "integer", "array_max": maxsize},
-    {"name": "naics_codes", "type": "array", "array_type": "integer", "text_type": "search"},
+    {
+        "name": "naics_codes",
+        "type": "any",
+        "models": [
+            {
+                "name": "naics_codes",
+                "type": "object",
+                "min": 0,
+                "object_keys": {
+                    "require": {"type": "array", "array_type": "integer", "text_type": "search", "min": 0},
+                    "exclude": {"type": "array", "array_type": "integer", "text_type": "search", "min": 0},
```

Comment:
Is this for tiny shield validations? 
I see `"array_type": "integer"`, but aren't the naics_code arrays `string` arrays per the API docs?

Reply:
It turns out that TinyShield's integer type allows strings so long as they parse into integers.

Reply:
Closing loop: this is being kept this way because it allows tinyshield to verify the array items are in fact integers (even though they need to be passed in quotes).